### ISSUE TITLE
Error serializing linq IEnumerable

### ISSUE THREAD

**Issue by Tornhoof** [neuecc/Utf8Json#23]
Hi,
the following snippet throws with a Typeload Exception:
```
System.TypeLoadException occurred
  HResult=0x80131522
  Message=Type "Utf8Json.Formatters.System_Linq_Enumerable\+WhereSelectListIterator`2\[\[WhereIteratorRepro_Test\, WhereIteratorRepro\]\,\[System_Guid\, mscorlib\]\]Formatter1" in der Assembly "Utf8Json.Resolvers.DynamicObjectResolverAllowPrivateFalseExcludeNullFalseNameMutateOriginal, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null" tries, eine Schnittstelle zu implementieren, auf die nicht zugegriffen werden kann.
  Source=Utf8Json
  StackTrace:
   at Utf8Json.JsonFormatterResolverExtensions.GetFormatterWithVerify[T](IJsonFormatterResolver resolver)
   at Utf8Json.JsonSerializer.Serialize[T](T value, IJsonFormatterResolver resolver)
   at Utf8Json.JsonSerializer.NonGeneric.Serialize(Type type, Object value, IJsonFormatterResolver resolver)
   at Utf8Json.JsonSerializer.NonGeneric.Serialize(Object value)
```
Sorry about the german exception but the error is basically typeloadexception for an inaccessible interface.

```csharp
using System;
using System.Collections.Generic;
using System.Linq;

namespace WhereIteratorRepro
{
    class Program
    {
        static void Main(string[] args)
        {

            var list = new List<Test>
            {
                new Test {Value = "Hello World", Id = Guid.NewGuid()},
                new Test {Value = "Hello Universe", Id = Guid.NewGuid()}
            };

            var select = list.Select(a => a.Id);
            var serialized = Utf8Json.JsonSerializer.NonGeneric.Serialize(select); // throws
            
        }

    }

    public class Test
    {
        public Guid Id { get; set; }
        public string Value { get; set; }
    }
}

```

**Comment by neuecc** [comment]
Thank you, I've understand what's happening.

It try to serialize `System.Linq.Enumerable+SelectListIterator`2` caused from NonGeneric(search concrete  type) but it can not access from outer(internal).

DynamicGenericResolver handles `IEnumerable<T>` inherit types before try to create dynamic formatter.
https://github.com/neuecc/Utf8Json/blob/master/src/Utf8Json/Resolvers/DynamicGenericResolver.cs#L252-L269
It is a simple solution, but it makes it impossible to decilize unique classes that inherit a lot of `IEnumerable<T>`.

Therefore, it responds by generating special code for `IEnumerable<T>` in DynamicObjectResolver....

**Comment by neuecc** [comment]
I've released `v1.3.0` it includes this fix.

**Comment by Tornhoof** [comment]
Thank you, it works for me. Closing this.
